src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryPublisher.cs

```diff
@@ -41,6 +41,11 @@ public ValueTask PublishEvent(EvaluationEvent evaluationEvent, CancellationToken
                 properties["Variant"] = evaluationEvent.Variant.Name;
             }
 
+            if (evaluationEvent.AssignmentReason != AssignmentReason.None)
```

Comment:
I want to call out that the ApplicationInsightsTelemetryPublisher will not send AssignmentReason telemetry if the AssignmentReason is None (which will only occur when the variant is null).

Reply:
Then it doesn't sound right. We should send the telemetry. This is a case people will want to know that the feature flag is misconfigured.

Reply:
> This is a case people will want to know that the feature flag is misconfigured.

I'm not sure what you intend to be discovered here. What would you expect "AssignmentReason" to be on a basic on/off feature flag? We already don't send variant in this case. I'm not sure what meaning "AssignmentReason" would have if no variant has been assigned.

Reply:
Sync'ed up with Jimmy offline and Jimmy help clarify the scenario. I was confused by the comment for `None`: _No variant is assigned_. `None` actually indicates the variant evaluation never occurs so the assignment is irrelevant. The assignment reason more like "NA" in this case. We could update the comment to make it clearer.